metrics/test_driver/cli/common/command/command_builder.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:meta/meta.dart';
+
+/// Base class for the command wrappers.
+///
+/// Defines the base methods to wrap a command.
+abstract class CommandBuilder {
+  final List<String> _args = [];
+
+  /// Creates the [List] of args from the [CommandBuilder] class instance.
+  List<String> buildArgs() => List.unmodifiable(_args);
+
+  /// Adds the [arg] to the argument list.
+  @protected
+  void add(String arg) {
```

Comment:
What do you think about adding @nonVirtual annotation to ensure that implementers won't override add() and addAll() methods?